models/linear_uncertainty.py

```diff
@@ -39,7 +40,7 @@ def update(self, observation, action, reward=0.0):
 
             self.xy_prods += reward * observation.squeeze()
             matvec_prod = self.covariance_matrix[a] @ observation.transpose(-1, -2)
-            scale = 1 + observation @ matvec_prod
+            scale = self.likelihood_variance + observation @ matvec_prod
```

Comment:
I think this likelihood variance needs to go outside the update, not instead of the one?

Reply:
Please let me explain how I get there.. 

I started with recursive update of covariance matrix:
$\Sigma_{t+1} = [\zeta \Sigma_{t}^{-1} + \beta^{-1} \phi \phi^\top]^{-1}$

Then using Matrix Inversion Lemma to get:
$\Sigma_{t+1} = \zeta^{-1} \Sigma_t - \zeta^{-2} \Sigma_t \phi \phi^\top \Sigma_t / (\beta + \zeta^{-1} \phi^\top \Sigma_t \phi)$

Then using \zeta=1.0, I ended up with the code above.
Sorry, it is a bit hard to read... not sure how to write equations here. Hope it makes sense, if not, please let me know.

Many thanks,
Taku